javers-core/src/main/java/org/javers/core/commit/CommitMetadata.java

```diff
@@ -5,17 +5,20 @@
 import org.joda.time.LocalDateTime;
 
 import java.io.Serializable;
+import java.util.Map;
 
 public class CommitMetadata implements Serializable {
 
     private final String author;
+    private final Map<String, String> properties;
     private final LocalDateTime commitDate;
-    private CommitId id;
+    private final CommitId id;
 
-    public CommitMetadata(String author, LocalDateTime commitDate, CommitId id) {
-        Validate.argumentsAreNotNull(author, commitDate, id);
+    public CommitMetadata(String author, Map<String, String> properties, LocalDateTime commitDate, CommitId id) {
+        Validate.argumentsAreNotNull(author, properties, commitDate, id);
 
         this.author = author;
+        this.properties = properties;
```

Comment:
do this.properties kopia tego co dał użytkownik, chcemy żeby commit był maksymalnie immutable